Swiftz/Reader.swift

```diff
@@ -0,0 +1,149 @@
+//
+//  Reader.swift
+//  Swiftz
+//
+//  Created by Matthew Purland on 11/25/15.
+//  Copyright © 2015 TypeLift. All rights reserved.
+//
+
+import Foundation
+
+/// A type for a `Reader` monad
+public protocol ReaderType {
```

Comment:
They definitely are.  But what I want is to introduce them all at once into a future major version of the framework.  For now, I think we should hold off on introducing them into just one place and plan for bigger things.